aten/src/ATen/test/vmap_test.cpp

```diff
@@ -850,4 +850,217 @@ TEST(VmapTest, TestBatchedTensorPermute) {
   }
 }
 
+static void checkMultiBatchVmapTransform(TensorList inputs, TensorList expected_outputs) {
+  auto outputs = MultiBatchVmapTransform::logicalToPhysical(inputs);
+  ASSERT_EQ(outputs.size(), expected_outputs.size());
+  for (int64_t idx = 0; idx < outputs.size(); idx++) {
+    const auto& output = outputs[idx].tensor();
+    ASSERT_EQ(output.data_ptr(), expected_outputs[idx].data_ptr());
+    ASSERT_EQ(output.sizes(), expected_outputs[idx].sizes());
+    ASSERT_TRUE(at::allclose(output, expected_outputs[idx]));
+  }
+}
+
+TEST(VmapTest, TestMultiBatchVmapTransformBatchedBatched) {
+  {
+    // Check that batch dims get moved to the front
+    int64_t B0 = 5, B1 = 7;
+    Tensor x = at::randn({2, B0, 3, B1});
+    Tensor y = at::randn({B1, 2, 3, B0});
+    Tensor batched_x = makeBatched(x, {{0, 1}, {1, 3}});
```

Comment:
Not the fault of this PR, but I get a pretty bad case of argument blindness with `{{0, 1}, {1, 3}}` here (takes me a moment to translate this into "batch dim 0 at position 1; batch dim 1 at position 3".

Reply:
I see later expands that look like this lol

Reply:
Sorry that is just me being lazy. I get argument blindness over that too. I've changed all instances of e.g. `{{0, 1}, {1, 3}}` to have comments that show what the dims are:
```
{{/*lvl*/0, /*dim*/3}, {/*lvl*/1, /*dim*/3}}
```